runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2231,7 +2370,7 @@ J9::ValuePropagation::doDelayedTransformations()
       else
          {
          TR_ASSERT_FATAL(isStore && isFlattenedElement, "Missing flags: isLoad %d isStore %d isFlattenedElement %d for call tree n%dn\n", isLoad, isStore, isFlattenedElement, callTree->getNode()->getGlobalIndex());
-         isCallTreeRemoved = transformFlattenedArrayElementStore(callToTransform->_arrayClass, callTree, callNode, needsNullValueCheck);
+         isCallTreeRemoved = transformFlattenedArrayElementStore(callToTransform->_arrayClass, callTree, callNode, needsNullStoreCheck);
```

Comment:
Could you also replace `needsNullValueCheck` in the declaration and definition of `transformFlattenedArrayElementStore` to the new name to be consistent? 

I wonder if the flag `RequiresNullValueCheck` should be updated to `RequiresNullStoreCheck` as well to make it consistent

Reply:
Sorry about that inconsistency!  I started off prototyping the split of `RequiresStoreAndNullCheck` into two flags using the names `RequiresStoreCheck` and `RequiresNullStoreCheck`, and then didn't quite merge things in a consistent fashion with your naming of `RequiresNullValueCheck`.  There's no reason to change the name of the flag, so I'll adjust the names of these boolean variables.

Reply:
Adjusted naming in commit 3e3941b73798c35167fcb8da13ea729159c3914d